editor/plugins/control_editor_plugin.cpp

```diff
@@ -792,230 +771,300 @@ void ControlEditorToolbar::_selection_changed() {
+	p_separator->set_custom_minimum_size(Size2i(1, 1));
+	p_box->add_child(p_separator);
+}
 
-				if (enable_containers) {
-					container_h_presets_menu->set_disabled(false);
-					container_h_presets_menu->set_tooltip(TTR("Horizontal sizing setting for children of a Container node."));
-					container_v_presets_menu->set_disabled(false);
-					container_v_presets_menu->set_tooltip(TTR("Vertical sizing setting for children of a Container node."));
-				} else {
-					container_h_presets_menu->set_disabled(true);
-					container_h_presets_menu->set_tooltip(TTR("Children of regular controls are controlled by their anchors and offsets."));
-					container_v_presets_menu->set_disabled(true);
-					container_v_presets_menu->set_tooltip(TTR("Children of regular controls are controlled by their anchors and offsets."));
-				}
+void ContainerSizeFlagPicker::_flag_button_pressed(const Control::SizeFlags p_flag) {
+	int flags = p_flag;
+	if (expand_button->is_pressed()) {
+		flags |= Control::SIZE_EXPAND;
+	}
+
+	emit_signal("size_flags_selected", flags);
+}
+
+void ContainerSizeFlagPicker::_button_toggle_expand(bool p_pressed) {
+	if (p_pressed) {
+		expand_button->set_text(TTR("With Expand"));
+		expand_button->set_tooltip(TTR("Disable to only set Shrink/Fill flags."));
+	} else {
+		expand_button->set_text(TTR("Without Expand"));
+		expand_button->set_tooltip(TTR("Enable to also set the Expand flag."));
+	}
+}
+
+void ContainerSizeFlagPicker::set_allowed_flags(Vector<Control::SizeFlags> &p_flags) {
+	flag_buttons[Control::SIZE_SHRINK_BEGIN]->set_disabled(!p_flags.has(Control::SIZE_SHRINK_BEGIN));
+	flag_buttons[Control::SIZE_SHRINK_CENTER]->set_disabled(!p_flags.has(Control::SIZE_SHRINK_CENTER));
+	flag_buttons[Control::SIZE_SHRINK_END]->set_disabled(!p_flags.has(Control::SIZE_SHRINK_END));
+	flag_buttons[Control::SIZE_FILL]->set_disabled(!p_flags.has(Control::SIZE_FILL));
+
+	expand_button->set_disabled(!p_flags.has(Control::SIZE_EXPAND));
+	if (p_flags.has(Control::SIZE_EXPAND)) {
+		if (expand_button->is_pressed()) {
+			expand_button->set_tooltip(TTR("Disable to only set Shrink/Fill flags."));
+		} else {
+			expand_button->set_tooltip(TTR("Enable to also set the Expand flag."));
+		}
+	} else {
+		expand_button->set_pressed(false);
+		expand_button->set_tooltip(TTR("Some parents of the selected nodes do not support the Expand flag."));
+	}
+}
+
+void ContainerSizeFlagPicker::set_allowed_flags_all() {
+	flag_buttons[Control::SIZE_SHRINK_BEGIN]->set_disabled(false);
+	flag_buttons[Control::SIZE_SHRINK_CENTER]->set_disabled(false);
+	flag_buttons[Control::SIZE_SHRINK_END]->set_disabled(false);
+	flag_buttons[Control::SIZE_FILL]->set_disabled(false);
+
+	expand_button->set_disabled(false);
+	if (expand_button->is_pressed()) {
+		expand_button->set_tooltip(TTR("Disable to only set Shrink/Fill flags."));
+	} else {
+		expand_button->set_tooltip(TTR("Enable to also set the Expand flag."));
+	}
+}
```

Comment:
This method basically duplicates the previous one. Wouldn't it be better to just call `set_allowed_flags()` with all flags?

Reply:
A bit of duplication felt better than something like this in that one place that uses it:

```cpp
	Vector<Control::SizeFlags> allowed_all_flags;
	allowed_all_flags.push_back(Control::SIZE_SHRINK_BEGIN);
	allowed_all_flags.push_back(Control::SIZE_SHRINK_CENTER);
	allowed_all_flags.push_back(Control::SIZE_SHRINK_END);
	allowed_all_flags.push_back(Control::SIZE_FILL);
	allowed_all_flags.push_back(Control::SIZE_EXPAND);

	container_h_picker->set_allowed_flags(allowed_all_flags);
	container_v_picker->set_allowed_flags(allowed_all_flags);
```

But fair enough, I'll change it.

Reply:
`Vector` supports list initialization now.

Reply:
So something like this? Yeah, that's much nicer, thanks

```cpp
	Vector<Control::SizeFlags> allowed_all_flags = {
		Control::SIZE_SHRINK_BEGIN,
		Control::SIZE_SHRINK_CENTER,
		Control::SIZE_SHRINK_END,
		Control::SIZE_FILL,
		Control::SIZE_EXPAND,
	};
```